### ISSUE TITLE
[Suggestion] A Lightweight Visualization Option without LaTeX Dependency

### ISSUE THREAD

**Issue by Wangningsen** [xmed-lab/TAM#4]
Thank you for creating and sharing this fantastic work on TAM. It's an incredibly powerful and insightful tool for understanding MLLM behavior.

I'm writing to suggest a potential enhancement that could make this great project even more accessible to a wider audience.

## The Challenge:

While using the repository, I noticed that the text visualization relies on a texlive-xetex installation. On many shared servers or computing clusters, users (like myself) often don't have sudo privileges, which makes installing system-level packages like TexLive via apt-get difficult or impossible. This dependency, while enabling high-quality text rendering, might create a barrier to entry for some potential users.

## Proposal:

Would it be possible to offer a lightweight version or an alternative visualization mode that does not require a LaTeX environment?

For many use cases, the most critical piece of information is the activation heatmap on the image itself. A simplified output focusing solely on this visual map would still be immensely valuable and would make the project significantly easier to set up and run "out-of-the-box" in more environments.

Thank you for considering this suggestion. I believe it would greatly benefit the community.

**Comment by Wangningsen** [comment]
I've been thinking about how this could be implemented and have experimented with a possible solution.

My approach would be to:

- Isolate the visualization logic: Modify the multimodal_process function.

- Preserve the core calculations: The key is that all the advanced computation, including the excellent ECI (Estimated Causal Inference) logic for denoising, would remain completely intact. The change would only affect the final rendering step.

- Bypass LaTeX: Inside the modified multimodal_process, the calls to generate the text-based visualization via LaTeX would be skipped. The function would instead directly output the image with the heatmap overlay, which is the primary visual evidence.

This way, the scientific integrity of the activation map calculation is preserved, while the system dependency is removed.

I've managed to create a proof-of-concept with these changes, and it works well for **only image input(although it is quite clumsy)**. If you think this direction is promising, I would be more than happy to clean up my code and submit a Pull Request for your review.

I attach some images to illustrate the effect of my texlive-free implementation:

<img width="549" height="870" alt="Image" src="https://github.com/user-attachments/assets/931bcd98-b256-4f7d-b557-97452e756a2f" />

<img width="2152" height="868" alt="Image" src="https://github.com/user-attachments/assets/5f5ae240-ccb1-4baf-a696-caadf7d84e5c" />

I'm aware that my implementation is a humble attempt compared to your original, polished code, but I believe that providing a way to bypass the TexLive dependency would allow many more people to quickly get started and see the amazing results of your work.

**Comment by Eli-YiLi** [comment]
Thanks for your attention and valuable suggestions.

The text part is used to compare with vision activations to discover more insights. Besides, the topk tokens are also helpful.

Without latex, I previously realized a cv2 based txt visualization version, but it looks worse than latex. 

I will consider to add an option to vis img only if latex is not installed. 

Besides, another option is to save these logits and load in PC with latex to visualize it, although it is a little complex.

**Comment by Wangningsen** [comment]
That will help a lot. Thanks for your time!
